code/modules/events/false_alarm.dm

```diff
@@ -26,12 +27,18 @@
 	var/fake_allowed = 0
 	while (!fake_allowed)
 		if (E)
-			E.kill(0)
+			E.kill()
+			qdel(E)
+			E = null
 		EM = pick(EC.available_events)
-		E = new EM.event_type(EM)
+		E = new EM.event_type(EM,1)
 		fake_allowed = !E.no_fake
 
-	message_admins("False Alarm: [E]")
-	E.kill(0)
+	if (E.ic_name)
+		eventname = E.ic_name
+	else
+		eventname = EM.name
+
+	E.kill()
 	E.announce()
```

Comment:
As noted, the last definition of `E` could probably assigned to the GC here, just in case. While it should clean up properly, it's better to be safe than sorry.